client/task_runner.go

```diff
@@ -93,6 +93,12 @@ type TaskRunner struct {
 	// Must acquire persistLock when accessing
 	taskDirBuilt bool
 
+	// createdResources are all the resources created by the task driver
+	// across all attempts to start the task.
+	//
+	// Must acquire persistLock when accessing
```

Comment:
Quite odd to grab persist lock to access any field. Really the persist code should grab locks for what it is writing to disk not the inverse